### ISSUE TITLE
input dimension

### ISSUE THREAD

**Issue by sepidhk** [ZFTurbo/segmentation_models_3D#14]
my input data has the shape of (118, 224, 224, 35), 118 3D nii images, width: 224, height: 224 and depth (#of slices) 35. My question what is the correct order of input shape? is the order of (#images, width, height, #slices) correct?

**Comment by ZFTurbo** [comment]
Shape must be (#images, width, height, #slices, 1). Last 1 is needed.

**Comment by sepidhk** [comment]
I have followed the recommended input shape and getting error of **ValueError: axes don't match array** or     **ValueError: Input 0 of layer "model_3" is incompatible with the layer: expected shape=(None, 96, 224, 224, 35), found shape=(1, 224, 224, 35, 1)**.

**Comment by ZFTurbo** [comment]
You put input shape:
`input_shape=(96, 224, 224, 35)`

Do you have 35 channels? 

If you have 1 channel use:

`input_shape=(224, 224, 35, 1)`

**Comment by ZFTurbo** [comment]
Shape must be: `(width, height, depth, channels)`. What is 96?

**Comment by sepidhk** [comment]
96 is the number of images.

**Comment by ZFTurbo** [comment]
You don't need to specify it. `None` is used for this. It means it can be any for model

**Comment by sepidhk** [comment]
Thanks for the reply. Now with Linknet and densenet121 I have this error: **ValueError: Inputs have incompatible shapes. Received shapes (56, 56, 8, 128) and (56, 56, 9, 128)** and  with UNet and densenet121 , I have the following error: **ValueError: A `Concatenate` layer requires inputs with matching shapes except for the concatenation axis. Received: input_shape=[(None, 56, 56, 8, 128), (None, 56, 56, 9, 128)]** - Is this because of the depth (#slices) not divisable by 32?

**Comment by ZFTurbo** [comment]
Yes. Each dimension must be divisible on 32. I think I should add warning message for this case.

**Comment by ZFTurbo** [comment]
I need to think how to make it work for any value of dimensions.

**Comment by menari123** [comment]
Yes, I have the same problem. I need manipule my image slices to use this model :/

**Comment by Ek-Ram** [comment]
Hi, 
I got the error : ValueError: axes don't match array. My input shape is  (128,128,128, 4).
model = sm.Unet(
    'resnet50',
    input_shape=(128, 128, 128, 4),
    encoder_weights='imagenet',
    classes=3,
    activation='softmax'
)
is there any solution ?

**Comment by ZFTurbo** [comment]
Can you provide exact error message?

**Comment by Ek-Ram** [comment]
Yes,  the exact error message is below :
`Downloading data from https://github.com/ZFTurbo/classification_models_3D/releases/download/v1.0.4/resnet50_inp_channel_3_tch_0_top_False.h5
185458688/185458584 [==============================] - 3s 0us/step
185466880/185458584 [==============================] - 3s 0us/step
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
[<ipython-input-15-b6141e2f92a6>](https://localhost:8080/#) in <cell line: 2>()
      1 #Fit the model
----> 2 model = sm.Unet(
      3     'resnet50',
      4     input_shape=(128, 128, 128, 4),
      5     encoder_weights='imagenet',

10 frames
[/usr/local/lib/python3.10/dist-packages/segmentation_models_3D/__init__.py](https://localhost:8080/#) in wrapper(*args, **kwargs)
     32         kwargs['models'] = _KERAS_MODELS
     33         kwargs['utils'] = _KERAS_UTILS
---> 34         return func(*args, **kwargs)
     35 
     36     return wrapper

[/usr/local/lib/python3.10/dist-packages/segmentation_models_3D/models/unet.py](https://localhost:8080/#) in Unet(backbone_name, input_shape, classes, activation, weights, encoder_weights, encoder_freeze, encoder_features, decoder_block_type, decoder_filters, decoder_use_batchnorm, dropout, **kwargs)
    224                          'Got: {}'.format(decoder_block_type))
    225 
--> 226     backbone = Backbones.get_backbone(
    227         backbone_name,
    228         input_shape=input_shape,

[/usr/local/lib/python3.10/dist-packages/segmentation_models_3D/backbones/backbones_factory.py](https://localhost:8080/#) in get_backbone(self, name, *args, **kwargs)
    100     def get_backbone(self, name, *args, **kwargs):
    101         model_fn, _ = self.get(name)
--> 102         model = model_fn(*args, **kwargs)
    103         return model
    104 

[/usr/local/lib/python3.10/dist-packages/classification_models_3D/models_factory.py](https://localhost:8080/#) in wrapper(*args, **kwargs)
    101             modules_kwargs = self.get_kwargs()
    102             new_kwargs = dict(list(kwargs.items()) + list(modules_kwargs.items()))
--> 103             return func(*args, **new_kwargs)
    104 
    105         return wrapper

[/usr/local/lib/python3.10/dist-packages/classification_models_3D/models/resnet.py](https://localhost:8080/#) in ResNet50(input_shape, input_tensor, weights, classes, stride_size, init_filters, include_top, repetitions, **kwargs)
    420         **kwargs
    421 ):
--> 422     return ResNet(
    423         MODELS_PARAMS['resnet50'],
    424         input_shape=input_shape,

[/usr/local/lib/python3.10/dist-packages/classification_models_3D/models/resnet.py](https://localhost:8080/#) in ResNet(model_params, input_shape, input_tensor, include_top, pooling, classes, stride_size, init_filters, weights, repetitions, **kwargs)
    338             model.load_weights(weights)
    339         else:
--> 340             load_model_weights(model, model_params.model_name,
    341                                weights, classes, include_top, **kwargs)
    342 

[/usr/local/lib/python3.10/dist-packages/classification_models_3D/weights.py](https://localhost:8080/#) in load_model_weights(model, model_name, dataset, classes, include_top, **kwargs)
     30         )
     31 
---> 32         model.load_weights(weights_path)
     33 
     34     else:

[/usr/local/lib/python3.10/dist-packages/keras/utils/traceback_utils.py](https://localhost:8080/#) in error_handler(*args, **kwargs)
     65     except Exception as e:  # pylint: disable=broad-except
     66       filtered_tb = _process_traceback_frames(e.__traceback__)
---> 67       raise e.with_traceback(filtered_tb) from None
     68     finally:
     69       del filtered_tb

/usr/local/lib/python3.10/dist-packages/numpy/core/overrides.py in transpose(*args, **kwargs)

[/usr/local/lib/python3.10/dist-packages/numpy/core/fromnumeric.py](https://localhost:8080/#) in transpose(a, axes)
    658 
    659     """
--> 660     return _wrapfunc(a, 'transpose', axes)
    661 
    662 

[/usr/local/lib/python3.10/dist-packages/numpy/core/fromnumeric.py](https://localhost:8080/#) in _wrapfunc(obj, method, *args, **kwds)
     55 
     56     try:
---> 57         return bound(*args, **kwds)
     58     except TypeError:
     59         # A TypeError occurs if the object does have such a method in its

ValueError: axes don't match array`

Thank you,

**Comment by ZFTurbo** [comment]
Try encoder_weights=None

**Comment by Ek-Ram** [comment]
I tried encoder_weights=None, and it works.
but it initializes weight randomly. I prefer to use 'imagenet' weights. 
is there any solution?

**Comment by ZFTurbo** [comment]
Try to do this:

`pip install classification-models-3D==1.0.6`

**Comment by Ek-Ram** [comment]
Unfortunatly same error

> Try to do this:
> 
> `pip install classification-models-3D==1.0.6`

I got same error.
